roles/README.adoc

```diff
@@ -113,6 +113,16 @@ See also <<vars-vs-defaults,Vars vs Defaults>> which mandates "Avoid embedding l
 Since distribution-specific values are kind of "magic values", it applies to them.
 The same logic applies for providers: a role can load a provider-specific variable file, include a provider-specific task file, or both, as needed.
 Consider making paths to templates internal variables if you need different templates for different distributions.
+A role should have a single entry point. Requiring users to use `tasks_from: file.yml` is discouraged as this could cause important pre and post task files executed in `tasks/main.yml` to be skipped.
+The `state` or `action` keyword should be passed to a role when a role has such a capability and be required when calling the role. The `tasks/main.yml` should load the appropriate task file based on the action or state requested by the user, rather than the consumer having to have intimate knowledge of the file structure layout within the role.
+All facts added by a role should fall under a single fact key named as the role.
+All roles should use argspec validation, if multiple states or actions are supported argspec validation should be done with a task.
+All roles should use argspec validation at the egress to confirm facts are correctly set.
+The tasks folder or one of its subdirectories should not contain more than 10 files.
+All intermediate facts set during role execution should be prefaced with the role name.
+All roles should be packaged within an ansible collection
+Configuration management roles should be atomic, idempotent, and rollback when failures are encountered
```

Comment:
Should it also support dry-run capability?

Reply:
Yes, good catch. check-mode/dry run should be supported.......... I'm assuming there will be cases were the endpoint system makes it impossible, I think it would be best to call this out as a limitation of the endpoint not an omission from the content

Reply:
@ericzolf What I understand here is that we should remove this recommendation from this PR/section in favor of the work happening in PR 42.  Please let me know if I misunderstand and I can add it back in.